src/components/FeatureViews/Accounts/ProfileHeading.js

```diff
@@ -99,214 +113,343 @@ const GET_PROFILE_HEADING_QUERY = {
-					<BirthdayIcon className="iconScaling" />
-				</Grid>
-				<Grid className="rowPadding" item xs={5}>
-					<Typography className="rowText">{userInfo?.birthday}</Typography>
-				</Grid>
-			</>
-		);
+  const handleClosePasswordReset = () => {
+    setOpen(false);
+    setResetOpen(true);
+    const newPassword = generatePassword(8, false, /[\w\?\-]/);
+    setPassword(newPassword);
 
-		
-		
-			
+    resetPassword({
+      variables: {
+        password: newPassword,
+        userId: userInfo.user.id,
+      },
+    });
+  };
 
-		switch (accountType) {
-			case "student":
-				return (
-					<>
-						<IDRow />
-						<BirthdayRow />
-						<Grid className="rowPadding" item xs={1}>
-							<GradeIcon className="iconScaling" />
-						</Grid>
-						<Grid className="rowPadding" item xs={5}>
-							<Typography className="rowText">Grade {userInfo.grade}</Typography>
-						</Grid>
-						<PhoneRow />
-						<Grid className="rowPadding" item xs={1}>
-							<SchoolIcon className="iconScaling" />
-						</Grid>
-						<Grid className="rowPadding" item xs={5}>
-							<Typography className="rowText">{userInfo.school?.name}</Typography>
-						</Grid>
-						<EmailRow />
-					</>
-				);
-			case "INSTRUCTOR":
-				return (
-					<>
-						<IDRow width={12} />
-						<PhoneRow width={12} />
-						<EmailRow />
-					</>
-				);
-			case "PARENT":
-				return (
-					<>
-						<IDRow />
-						<Grid className="rowPadding" item xs={1}>
-							<MoneyIcon className="iconScaling" />
-						</Grid>
-						<Grid className="rowPadding" item xs={5}>
-							<Typography className="rowText">${userInfo.balance}</Typography>
-						</Grid>
-						<PhoneRow width={12} />
-						<EmailRow />
-					</>
-				);
-			default:
-				return (
-					<>
-						<IDRow width={12}/>
-						<PhoneRow width={12}/>
-						<EmailRow/>
-					</>
-				);
-		}
-	};
+  const handleCloseReset = () => {
+    setResetOpen(false);
+  };
+
+  const isAdmin = loggedInAuth.accountType === USER_TYPES.admin;
+  const isUser = loggedInAuth.user.id === userInfo.user.id;
+
+  const renderEditandAwayButton = () => (
+    <Grid container item xs={4}>
+      {(isAdmin || isUser) && (
+        <>
+          <Grid component={Hidden} item mdDown xs={12}>
+            <EditIcon className="editIcon" />
+            <div className="editResetDiv">
+              <ResponsiveButton
+                component={Link}
+                to={`/form/${userInfo.accountType.role}/${userInfo.user.id}`}
+                className="edit"
+              >
+                Edit Profile
+              </ResponsiveButton>
+              {isAdmin &&
+                (userInfo.accountType.role === "student" ? (
+                  <ResponsiveButton
+                    className="resetStudent"
+                    disabled
+                    onClick={handleClickOpen}
+                  >
+                    Reset Password
+                  </ResponsiveButton>
+                ) : (
```

Comment:
the only differences between these 2 components are the "className" and "disabled" simplify this code like this
```
//outside of return function
const isStudentProfile = userInfo.accountType.role === "student";
// inside component where you're adding `<ResponsiveButton/>`
<ResponsiveButton
    className={`reset${isStudentProfile && "Student"}
    disabled={isStudentProfile}
    onClick={handleClickOpen}
/>
```
This would replace lines 192 - 203

Reply:
Done thank you!